### PR TITLE
Backend/models

### LINKED ISSUE #6: Create all models using SQLize ORM

### PR DESCRIPTION
**Pull request by vutoan1245**
Resolves #6 
Also contains #26

### CODE REVIEW

File: backend/model/Folder.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const { DataTypes, Model } = require('sequelize')
```
Comment (kennygperez) [inline backend/model/Folder.ts:1]: Switch to import/export es6 syntax

File: backend/config/database.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Sequelize } from 'sequelize'
+
+module.exports = new Sequelize(
```
Comment (kennygperez) [inline backend/config/database.ts:3]: switch to export syntax

File: backend/src/folder.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        return {
+            statusCode: 400,
+            body: JSON.stringify({
+                message: 'Missing parameters',
+            }),
+        }
+    }
+
+    await Folder.sync()
+    return Folder.findAll({
```
Comment (kennygperez) [inline backend/src/folder.ts:20]: Using promises and async/await in the same function choose the async/await pattern

File: backend/src/folder.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        return {
+            statusCode: 400,
+            body: JSON.stringify({
+                message: 'Missing parameters',
+            }),
+        }
+    }
+
+    await Folder.sync()
+    return Folder.create({
```
Comment (kennygperez) [inline backend/src/folder.ts:55]: Same here

File: backend/src/folder.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        return {
+            statusCode: 400,
+            body: JSON.stringify({
+                message: 'Missing parameters',
+            }),
+        }
+    }
+
+    await Folder.sync()
+    return Folder.update(
```
Comment (kennygperez) [inline backend/src/folder.ts:86]: Same here

File: backend/src/folder.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    if (!params?.folderId) {
+        return {
+            statusCode: 400,
+            body: JSON.stringify({
+                message: 'Missing parameters',
+            }),
+        }
+    }
+    await Folder.sync()
+    return Folder.destroy({ where: { id: params?.folderId } })
```
Comment (kennygperez) [inline backend/src/folder.ts:114]: And here

File: backend/src/folder.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
+import { APIGatewayEvent, Context, ProxyResult } from 'aws-lambda'
+const Folder = require('../model/Folder')
+const querystring = require('querystring')
```
Comment (kennygperez) [inline backend/src/folder.ts:3]: querystring is does what? is that native to the NodeJS?

**Review by kennygperez — CHANGES_REQUESTED** [review]
I would change the models like so in order to give us the most Typescript support

so for example change the `User.ts` file to the following

```ts
import { DataTypes, ModelDefined } from "sequelize";
import sequelize from "../config/database";

interface UserAttributes {
  firstName: string;
  lastName: string;
  toke: string;
  refreshToken: string;
}

interface UserCreationAttributes {
  firstName?: string;
  lastName?: string;
  toke?: string;
  refreshToken?: string;
}

export const User: ModelDefined<UserAttributes, UserCreationAttributes> =
  sequelize.define(
    "user",
    {
      firstName: {
        type: DataTypes.STRING,
      },
      lastName: {
        type: DataTypes.STRING,
      },
      token: {
        type: DataTypes.STRING,
      },
      refreshToken: {
        type: DataTypes.STRING,
      },
    },
    {
      tableName: "User",
    }
  );
```

File: backend/model/User.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { DataTypes, Model } from 'sequelize'
+const sequelize = require('../config/database')
+
+class User extends Model {}
+
+User.init(
+    {
```
Comment (kennygperez) [inline backend/model/User.ts:7]: I would also add a primary key to the User table

**Review by kennygperez — CHANGES_REQUESTED** [review]

**Comment by kennygperez** [comment]
I Would also add `allowNull: true` or `allowNull: false` to the column definitions also using primary keys

```
id: {
  type: DataTypes.INT,
  allowNull: false,
  primaryKey: true,
autoIncremenet: true
},
```

**Comment by kennygperez** [comment]
Another thing is the relationships between the tables were never defined. in a normal backend (non-lambda it would be in the index.ts of the model folder), I recommend creating a init lambda function that initializes the DB tables and relationships

File: backend/model/User.ts (hunk 8 above)
Comment (vutoan1245) [reply backend/model/User.ts:7]: sequelize adds a default id as a primary key.

File: backend/src/folder.ts (hunk 7 above)
Comment (vutoan1245) [reply backend/src/folder.ts:3]: Yes, it is native to NodeJS. It parses event.body in x-www-form-urlencoded format

**Review by kennygperez — CHANGES_REQUESTED** [review]
You are still using the class-based model definition, switch to the function-based model definition, and define the proper interfaces/types for accessibility for future devs

**Comment by kennygperez** [comment]
I'll fix the model pattern and types at a different time

**Review by kennygperez — APPROVED** [review]
Refer to my previous message

**Pull request merged**